python/cugraph/cugraph/sampling/heterogeneous_neighbor_sample.py

```diff
@@ -0,0 +1,349 @@
+
+    """
+
+
+    if compression not in ["COO", "CSR", "CSC", "DCSR", "DCSC"]:
+        raise ValueError("compression must be one of COO, CSR, CSC, DCSR, or DCSC")
+
+    if (
+        (compression != "COO")
+        and (not compress_per_hop)
+        and prior_sources_behavior != "exclude"
+    ):
+        raise ValueError(
+            "hop-agnostic compression is only supported with"
+            " the exclude prior sources behavior due to limitations "
+            "of the libcugraph C++ API"
+        )
+
+    if compress_per_hop and prior_sources_behavior != "carryover":
+        raise ValueError(
+            "Compressing the edgelist per hop is only supported "
+            "with the carryover prior sources behavior due to limitations"
+            " of the libcugraph C++ API"
+        )
+
+    if isinstance(start_list, int):
+        start_list = [start_list]
+
+    if isinstance(start_list, list):
+        start_list = cudf.Series(
+            start_list, dtype=G.edgelist.edgelist_df[G.srcCol].dtype
+        )
+    
+    if isinstance(starting_vertex_label_offsets, list):
+        starting_vertex_label_offsets = cudf.Series(
+            starting_vertex_label_offsets
+        )
+    
+    if isinstance(vertex_type_offsets, list):
+        vertex_type_offsets = cudf.Series(
+            vertex_type_offsets
+        )
+
+    # fanout_vals must be passed to pylibcugraph as a host array
+    if isinstance(fanout_vals, numpy.ndarray):
+        fanout_vals = fanout_vals.astype("int32")
+    elif isinstance(fanout_vals, list):
+        fanout_vals = numpy.asarray(fanout_vals, dtype="int32")
+    elif isinstance(fanout_vals, cp.ndarray):
+        fanout_vals = fanout_vals.get().astype("int32")
+    elif isinstance(fanout_vals, cudf.Series):
+        fanout_vals = fanout_vals.values_host.astype("int32")
+    else:
+        raise TypeError("fanout_vals must be a sequence, " f"got: {type(fanout_vals)}")
+
+    if "weights" in G.edgelist.edgelist_df:
+        weight_t = G.edgelist.edgelist_df["weights"].dtype
+    else:
+        weight_t = "float32"
+
+    start_list = ensure_valid_dtype(G, start_list)
+
+    if G.renumbered is True:
+        if isinstance(start_list, cudf.DataFrame):
+            start_list = G.lookup_internal_vertex_id(
+                start_list, start_list.columns
+            )
+        else:
+            start_list = G.lookup_internal_vertex_id(start_list)
+
+    if with_biases:
+        sampling_result_array_dict = pylibcugraph_heterogeneous_biased_neighbor_sample(
+            resource_handle=ResourceHandle(),
+            input_graph=G._plc_graph,
+            start_vertex_list=start_list,
+            starting_vertex_label_offsets=starting_vertex_label_offsets,
+            vertex_type_offsets=vertex_type_offsets,
+            h_fan_out=fanout_vals,
+            num_edge_types=num_edge_types,
+            with_replacement=with_replacement,
+            do_expensive_check=True,
```

Comment:
Should this be a parameter or at least hardcoded to `False`?

Reply:
I set it to True when debugging and didn't set it back to False. It is resolved now